modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -233,28 +253,30 @@ protected H2Tree(
         else {
             unwrappedPk = true;
 
-            inlineSize = unwrappedColsInfo.inlineSize();
+            cols = unwrappedCols.toArray(H2Utils.EMPTY_COLUMNS);
+
+            inlineIdxs = getAvailableInlineColumns(affinityKey, cacheName, idxName, log, pk,
+                table, cols, factory, true);
 
-            inlineIdxs = unwrappedColsInfo.inlineIdx();
+            inlineSize = computeInlineSize(inlineIdxs, configuredInlineSize, cctx.config().getSqlIndexMaxInlineSize());
 
             setIos(
                 H2ExtrasInnerIO.getVersions(inlineSize, mvccEnabled),
-                H2ExtrasLeafIO.getVersions(inlineSize, mvccEnabled));
-
-            initTree(initNew, inlineSize);
+                H2ExtrasLeafIO.getVersions(inlineSize, mvccEnabled)
+            );
         }
 
-        cols = unwrappedPk ? unwrappedColsInfo.cols() : wrappedColsInfo.cols();
-
         created = initNew;
+
+        initTree(created, inlineSize);
```

Comment:
I think more clearly will be keep the imvocation inside ``else`` block with ``true`` for first argument

Reply:
ok, I'll move it